src/aws-cpp-sdk-core/include/aws/core/utils/crypto/SecureRandom.h

```diff
@@ -4,6 +4,8 @@
  */
 #pragma once
 #include <type_traits>
+#include <memory>
```

Comment:
why?

Reply:
non-unity builds fail because they are missing a include for `std::shared_ptr`, it would appear that this use to get that include transitively before from crypto impl, but now it no longer does.

Reply:
we wrap shared ptr, shouldnt we include that instead?

Reply:
we wrap `make_shared` but not actually `shared_ptr` i.e where we [define make shared](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/include/aws/core/utils/memory/stl/AWSAllocator.h#L104)

```
std::shared_ptr<T> MakeShared(const char* allocationTag, 
  ArgTypes&&... args)
{
  return std::allocate_shared<T, Aws::Allocator<T>>(
    Aws::Allocator<T>(), std::forward<ArgTypes>(args)...);
}
```

we actually return a std pointer with a allocator, the only reason a unique wrapper exists is because [`make_unique`](https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique) slipped into c++14 and not 11....which still frustrates me